roles/ceph-mon/tasks/deploy_monitors.yml

```
@@ -158,13 +158,13 @@
   ansible.builtin.command: >
     {{ ceph_monmaptool_cmd }}
     --create
-    {% for host in _monitor_addresses -%}
+    {% for name, addr in _monitor_addresses.items() -%}
     --addv
-    {{ host.name }}
+    {{ name }}
```

This needs to be the hostname as the `inventory_hostname` might be a fqdn
```suggestion
    {{ hostvars[name]['ansible_hostname'] }}
```